File: arrays and hashes/431.js
// # Convert an array of arrays into a hash.
// # For example, [[1, 3], [8, 9], [2, 16]] becomes {1 => 3, 8 => 9, 2 => 16}.

var a = [
  [1, 3],
  [8, 9],
  [2, 16],
];

var i = 0;
var hash = {};
while (i < a.length) {
  var key = a[i][0];
  var value = a[i][1];
  hash[key] = value;
  i += 1;
}
console.log(hash);

File: arrays and hashes/431.rb
# Convert an array of arrays into a hash.
# For example, [[1, 3], [8, 9], [2, 16]] becomes {1 => 3, 8 => 9, 2 => 16}.

a = [[1, 3], [8, 9], [2, 16]] 

i = 0
hash = {}
while i < a.length
  key = a[i][0]
  value = a[i][1]
  hash[key] = value
  i += 1
end 
p hash